src/GraphQL/Types/ISchema.cs

```diff
@@ -34,6 +34,13 @@ public interface ISchema : IProvideMetadata
         /// </summary>
         INameConverter NameConverter { get; }
 
+        /// <summary>
+        /// Note that field middlewares apply only to an uninitialized schema. If the schema is initialized
+        /// then adding additional middleware through the builder does nothing. The schema is initialized (if not yet)
+        /// at the beginning of the first call to <see cref="DocumentExecuter"/>.<see cref="DocumentExecuter.ExecuteAsync(ExecutionOptions)">ExecuteAsync</see>.
```

Comment:
```suggestion
        /// at the beginning of the first call to <see cref="DocumentExecuter"/>.<see cref="DocumentExecuter.ExecuteAsync(ExecutionOptions)">ExecuteAsync</see>.
        /// However, you can also apply middlewares at any time in runtime using SchemaTypes.ApplyMiddleware method.
```

Also let's make `ApplyMiddleware` method syncronized - lock or `[MethodImpl(MethodImplOptions.Synchronized)]`.

Reply:
I don't see a reason for that, unless we lock executions from happening simultaneously via `ReaderWriterLockSlim`, which I wouldn't do.  If someone is adding middleware at runtime, it's their responsibility to be sure that there are no active executions or other threads also adding middleware.

Reply:
Agree. Yes, I thought about it yesterday too. Middleware is installed at the same time as the executing request, so the result is still undefined. It was very late and in a lot of reviews I just forgot to write about it.